rollbar-java/src/main/java/com/rollbar/notifier/sender/SyncSender.java

```diff
@@ -0,0 +1,136 @@
+package com.rollbar.notifier.sender;
+
+import com.rollbar.api.payload.Payload;
+import com.rollbar.notifier.sender.exception.SenderException;
+import com.rollbar.notifier.sender.json.JsonSerializer;
+import com.rollbar.notifier.sender.json.JsonSerializerImpl;
+import com.rollbar.notifier.sender.result.Result;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Synchronous implementation of the {@link Sender sender}.
+ */
+public class SyncSender implements Sender {
+
+  public static final String DEFAULT_API_ENDPOINT = "https://api.rollbar.com/api/1/item/";
+
+  private static final String UTF_8 = StandardCharsets.UTF_8.name();
+
+  private final URL url;
+
+  private final JsonSerializer serializer;
+
+  /**
+   * Constructor.
+   */
+  public SyncSender() {
+    this(DEFAULT_API_ENDPOINT);
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param url the Rollbar API endpoint.
+   */
+  public SyncSender(String url) {
+    this(url, new JsonSerializerImpl());
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param url the Rollbar API endpoint.
+   * @param serializer the serializer.
+   */
+  public SyncSender(String url, JsonSerializer serializer) {
+    try {
```

Comment:
Should this call the constructor below?

Reply:
I'll take a look at it because the approach we are following is based on having builders to not call directly the constructors.